### ISSUE TITLE
Prevent Checkpoints Overlap

### ISSUE THREAD

**Issue by NewAlexGamer** [absolllute/Mega-Hack-Pro-Future#991]
have you ever practiced the new top 1 just to see how hard it was? well, i'm pretty sure you've gone through this too. when trying to pass an obstacle you've probably spammed the click and checkpoint button but realized it was not going to work. so you decide to go back, but wait, pressing 'x' doesn't do anything, why is that? well lemme tell ya, you've placed about 15 checkpoints in literally the same frame so you now have to press 'x' 14 more times, but you don't know it's exactly 14. so after spamming to remove them you realize half a second too late that you need to stop and in the meanwhile you removed the checkpoints of about 5% more of the level, which for a top 1 is worse than death. now you're depressed and go in the corner of the room contemplating why checkpoints overlap

**Comment by MainConMan** [comment]
Yes I need this to be a thing

**Comment by NefariousIntent2** [comment]
> have you ever practiced the new top 1 just to see how hard it was? well, i'm pretty sure you've gone through this too. when trying to pass an obstacle you've probably spammed the click and checkpoint button but realized it was not going to work. so you decide to go back, but wait, pressing 'x' doesn't do anything, why is that? well lemme tell ya, you've placed about 15 checkpoints in literally the same frame so you now have to press 'x' 14 more times, but you don't know it's exactly 14. so after spamming to remove them you realize half a second too late that you need to stop and in the meanwhile you removed the checkpoints of about 5% more of the level, which for a top 1 is worse than death. now you're depressed and go in the corner of the room contemplating why checkpoints overlap

I had this up to 28% of the level. Yay!

**Comment by NefariousIntent2** [comment]
> have you ever practiced the new top 1 just to see how hard it was? well, i'm pretty sure you've gone through this too. when trying to pass an obstacle you've probably spammed the click and checkpoint button but realized it was not going to work. so you decide to go back, but wait, pressing 'x' doesn't do anything, why is that? well lemme tell ya, you've placed about 15 checkpoints in literally the same frame so you now have to press 'x' 14 more times, but you don't know it's exactly 14. so after spamming to remove them you realize half a second too late that you need to stop and in the meanwhile you removed the checkpoints of about 5% more of the level, which for a top 1 is worse than death. now you're depressed and go in the corner of the room contemplating why checkpoints overlap

Also how does this work, what if you really need to spam checkpoints (wave spam)

**Comment by sinc0s** [comment]
Good idea. Also might be a good idea to fix other things with checkpoints, such as them disappearing behind hitbox player trail.

**Comment by MainConMan** [comment]
Invisible checkpoint bug fix was suggested in #404 but the one about them not being hidden by the hitbox is a good idea!
